### PR TITLE
fix: merge prefix maps from configuration and result

### LINKED ISSUE #153: new namespaces of endpoint are lost
When converting a model to json, only the prefixes of the configuration are kept. If the endpoint created new prefixes, these are lost.

### PR DESCRIPTION
**Pull request by michielrogissart-cognizone**

### CODE REVIEW

File: cube/src/main/java/zone/cogni/asquare/cube/convertor/ModelToJsonConversion.java [hunk 1]
```diff
@@ -205,14 +205,30 @@ private boolean modelContainsRoot(Model model, Resource subject) {
...
 
     ObjectNode contextNode = context.jsonRoot.putObject("context");
     ObjectNode prefixNode = contextNode.putObject("prefix");
     prefixes.forEach(prefixNode::put);
   }
 
+  private Map<String, String> mergePrefixMaps(Map<String, String> map1, Map<String, String> map2) {
+    Map<String, String> result = new HashMap<>(map1);
+
+    result.putAll(
```
Comment (natan-cox-cognizone) [inline cube/src/main/java/zone/cogni/asquare/cube/convertor/ModelToJsonConversion.java]: not sure if creating a new map and then invoking putAll is good: it is more lines of code and more CPU cycles.

File: cube/src/main/java/zone/cogni/asquare/cube/convertor/ModelToJsonConversion.java [hunk 2]
```diff
@@ -205,14 +205,30 @@ private boolean modelContainsRoot(Model model, Resource subject) {
...
+            map2.entrySet().stream()
+                    .filter(e -> !map1.containsValue(e.getValue()))
+                    .map(e -> map1.containsKey(e.getKey()) ? Map.entry(e.getKey()+"0", e.getValue()) : e)
```
Comment (natan-cox-cognizone) [inline cube/src/main/java/zone/cogni/asquare/cube/convertor/ModelToJsonConversion.java]: again as before: + "0" only works if "prefix0" does not exist yet

**Review by natan-cox-cognizone — CHANGES_REQUESTED** [review]

File: cube/src/main/java/zone/cogni/asquare/cube/convertor/ModelToJsonConversion.java [hunk 3]
```diff
@@ -205,14 +205,32 @@ private boolean modelContainsRoot(Model model, Resource subject) {
...
+                                                    .filter(e -> !map1.containsValue(e.getValue()))
+                                                    .map(e -> {
+                                                      if (!map1.containsKey(e.getKey())) return e;
+                                                      int i = 0;
+                                                      while (map1.containsKey(e.getKey()+i)) i++;
+                                                      return Map.entry(e.getKey()+i, e.getValue());
+                                                    });
+
+
+    return Stream.concat(map1.entrySet().stream(), map2FilteredStream).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
```
Comment (natan-cox-cognizone) [inline cube/src/main/java/zone/cogni/asquare/cube/convertor/ModelToJsonConversion.java]:
```suggestion
    return Stream.concat(map1.entrySet().stream(), map2FilteredStream)
                 .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
```

one statement per line

File: cube/src/main/java/zone/cogni/asquare/cube/convertor/ModelToJsonConversion.java [hunk 4]
```diff
@@ -205,14 +205,32 @@ private boolean modelContainsRoot(Model model, Resource subject) {
...
     ObjectNode contextNode = context.jsonRoot.putObject("context");
     ObjectNode prefixNode = contextNode.putObject("prefix");
     prefixes.forEach(prefixNode::put);
   }
 
+  private Map<String, String> mergePrefixMaps(Map<String, String> map1, Map<String, String> map2) {
+    Stream<Map.Entry<String, String>> map2FilteredStream = map2.entrySet().stream()
+                                                    .filter(e -> !map1.containsValue(e.getValue()))
+                                                    .map(e -> {
+                                                      if (!map1.containsKey(e.getKey())) return e;
```
Comment (natan-cox-cognizone) [inline cube/src/main/java/zone/cogni/asquare/cube/convertor/ModelToJsonConversion.java]: maybe a separate method? like newKeyEntry(...)

Comment (michielrogissart-cognizone) [reply cube/src/main/java/zone/cogni/asquare/cube/convertor/ModelToJsonConversion.java]: I'll do that, I wasn't sure if it was needed to make a function for something this small. It does look kind of ugly now :/

**Review by natan-cox-cognizone — APPROVED** [review]

**Pull request merged**